src/test/java/codeu/controller/SearchServletTest.java

```diff
@@ -0,0 +1,62 @@
+package codeu.controller;
+
+import codeu.model.data.User;
+import codeu.model.store.basic.UserStore;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.PrintWriter;
+import java.time.Instant;
+import java.util.UUID;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import com.google.gson.Gson;
+
+public class SearchServletTest {
+
+  private SearchServlet searchServlet;
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private UserStore mockUserStore;
+
+  @Before
+  public void setup() {
+    searchServlet = new SearchServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    
+    mockUserStore = Mockito.mock(UserStore.class);
+    searchServlet.setUserStore(mockUserStore);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    List<User> fakeUserList = new ArrayList<User>();
+    fakeUserList.add(new User(UUID.randomUUID(), "user", "password_hash", Instant.now()));
+    Mockito.when(mockUserStore.getUsers("user")).thenReturn(fakeUserList);
+
+    String json = new Gson().toJson(fakeUserList);
+
+    Mockito.when(mockRequest.getParameter("searchRequest")).thenReturn("user");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    Mockito.when(mockResponse.getWriter()).thenReturn(writer);
+
+    searchServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.when(mockResponse.getContentType()).thenReturn("application/json");
+    Mockito.when(mockResponse.getCharacterEncoding()).thenReturn("UTF-8");
+
+    Assert.assertEquals("application/json", mockResponse.getContentType());
```

Comment:
I'm curious why lines 55-56 are needed for lines 58-59 to work? I still can't completely wrap my head around this.

Reply:
I'm not sure exactly why you're calling getContentType() and getCharacterEncoding() on the response object. If you want to verify that it calls the setter methods correctly with the correct arguments, you could just use argument captor.

Reply:
It confused me for a while too, but once I found [this SO question](https://stackoverflow.com/questions/50403688/return-of-getcontenttype-in-mocked-httpservletresponse-instance-is-null) I thought, of course! You're not using an actual `HttpServletResponse` object here, so the `setContentType` call doesn't really do anything.

Reply:
That link cleared things up for me, thanks. I tried using an argument captor but I understood the link example more.